File: android/app/src/main/kotlin/com/typography/animations/entrance/ElasticStretch.kt
package com.typography.animations.entrance

import com.typography.AnimatedTextState
import com.typography.EasingType
import com.typography.animations.EasingFunctions
import kotlin.math.cos

fun evaluateElasticStretch(t: Float): AnimatedTextState {
    val easeT = EasingFunctions.applyEasing(t, EasingType.EASE_OUT)
    val elasticT = EasingFunctions.applyEasing(t, EasingType.ELASTIC_OUT)
    
    val offsetY = -0.3f * (1.0f - elasticT)
    val scaleX = 1.0f - 0.4f * (1.0f - easeT) * cos(t * 3.14159265f)
    val scaleY = 1.0f + 0.6f * (1.0f - easeT) * cos(t * 3.14159265f)
    
    return AnimatedTextState(
        opacity = t.coerceIn(0f, 1f),
        offsetY = offsetY,
        scaleX = scaleX,
        scaleY = scaleY
    )
}

File: lib/utils/animations/entrance/elastic_stretch.dart
import 'dart:math';
import '../../base/animated_text_state.dart';
import '../../easing/easing_functions.dart';
import '../../../models/editor_models.dart';

AnimatedTextState evaluateElasticStretch(double t) {
  final easeT = EasingFunctions.applyEasing(t, EasingType.easeOut);
  final elasticT = EasingFunctions.applyEasing(t, EasingType.elasticOut);
  
  final offsetY = -0.3 * (1.0 - elasticT);
  final scaleX = 1.0 - 0.4 * (1.0 - easeT) * cos(t * pi);
  final scaleY = 1.0 + 0.6 * (1.0 - easeT) * cos(t * pi);
  
  return AnimatedTextState(
    opacity: t.clamp(0.0, 1.0),
    offsetY: offsetY,
    scaleX: scaleX,
    scaleY: scaleY,
  );
}
